Enable lpit for mke17z

FelixWang47831 opened the pull request:
Enable LPIT on MKE17Z, including:
1. devicetree enablement
2. clock config
3. counter test case enablement
4. Fix LPIT disable interrupt bug

FelixWang47831 committed:
dts: arm: nxp: ke1xz: add lpit instances for KE1XZ

Add lpit0 and all of channels settings

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

FelixWang47831 committed:
soc: nxp: kinetis: clock update for LPIT instances on KE1XZ

Configure LPIT0 IP clock if devicetree status is okay.

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

FelixWang47831 committed:
tests: drivers: counter_basic_api: Enable lpit test for frdm_ke17z

Enable test for all channels.
This test covers following test cases:
- test_set_top_value_without_alarm
- test_set_top_value_with_alarm
- test_valid_function_without_alarm

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

decsny commented on tests/drivers/counter/counter_basic_api/boards/frdm_ke17z.overlay:
```diff
@@ -0,0 +1,25 @@
+	status = "okay";
+};
+
+&lpit0_channel1 {
+	status = "okay";
+};
```
i like to leave one of the channels disabled to make sure the macrobatics doesnt break

FelixWang47831 replied:
Sure, updated.

decsny commented on tests/drivers/counter/counter_basic_api/boards/frdm_ke17z.conf:
delete this file, it's not necessary, the config gets turned on if the node is okay in DT. generally do not manually set driver configs or add files that are not necessary.

FelixWang47831 replied:
Updated.

FelixWang47831 committed:
boards: nxp: frdm_ke17z: enable lpit driver

Enable lpit0 for frdm_ke17z

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

FelixWang47831 committed:
drivers: Counter: Fix LPIT disable interrupt bug

The parameter for LPIT_DisableInterrupts should be
lpit_interrupt_enable_t, not channel id.

Besides, improve the code to use BIT macro for shift
operation.

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

FelixWang47831 commented on drivers/counter/counter_mcux_lpit.c:
```diff
@@ -75,7 +75,7 @@ static int mcux_lpit_stop(const struct device *dev)
 	uint8_t channel_id = LPIT_CHANNEL_ID(dev);
 
-	LPIT_DisableInterrupts(config->base, channel_id);
+	LPIT_DisableInterrupts(config->base, (1U << channel_id));
```
Fixed.
